Assets/Scripts/MovePlayer.cs

```diff
@@ -47,32 +59,40 @@ void Update()
             }
         }
         Vector2 movement = new Vector2(deltaX, rgb.velocity.y);
-        rgb.velocity = movement;
-        Vector2 max = boxCollider.bounds.max;
-        Vector2 min = boxCollider.bounds.min;
-        Vector2 maxForDetect = new Vector2(max.x, min.y - 0.01f);
-        Vector2 minForDetect = new Vector2(min.x, min.y - 0.02f);
-        Collider2D hit = Physics2D.OverlapArea(minForDetect, maxForDetect,LayerMask.GetMask("Ground"));//�ж�����Ƿ��ڵ�����
-        //Physics2D.gravity = new Vector3(0, gravity, 0);
-        bool grounded = false;
-        if (hit != null && hit.gameObject.tag == "Ground")
+        rgb.velocity = movement;//Ϊ��������ٶ�
+        jumpDirection direction = getAllowJumpDirection();//��ȡ���������������Ծ�ķ���
+        if(Input.GetKeyDown(KeyCode.W))//��Ұ�����Ծ��
         {
-            grounded = true;
-        }
-        //Debug.Log("grounded : " + i+++ " " + grounded);
-        if (grounded && Input.GetKeyDown(KeyCode.W))
-        {
-            animator.SetBool("BackToIdle", false);
-            animator.SetBool("FinishJump", false);
-            animator.SetBool("PrepareJump", true);
-            StartCoroutine(Jump());
+            if(direction == jumpDirection.Up)//��Ծ�����������
+            {
+                animator.SetBool("BackToIdle", false);//���ò�����Ծǰҡ����
+                animator.SetBool("FinishJump", false);
+                animator.SetBool("PrepareJump", true);
+                StartCoroutine(Jump());
+                lastJumpDirection = direction;
+            }
+            else if (direction == jumpDirection.Left&&direction!=lastJumpDirection)//��Ծ�����������
+            {
+                animator.SetBool("BackToIdle", false);
+                animator.SetBool("FinishJump", false);
+                rgb.velocity = new Vector2(rgb.velocity.x, jumpSpeed);
+                lastJumpDirection = direction;
+            }
+            else if(direction == jumpDirection.Right&&direction!=lastJumpDirection)//��Ծ�����������
+            {
+                animator.SetBool("BackToIdle", false);
+                animator.SetBool("FinishJump", false);
+                rgb.velocity = new Vector2(rgb.velocity.x, jumpSpeed);
+                lastJumpDirection = direction;
+            }
         }
-        if(isFalling==true&&grounded)
+        //Debug.Log(i++ + " " + "direction:"+direction);
+        if(isFalling==true&&direction==jumpDirection.Up)//�����������䲢���Ѿ����������˾Ͳ��Ž�����Ծ����
```

Comment:
isFalling这个变量没有任何存在的意义，直接使用rgb.velocity.y < 0f不可以吗？
```suggestion
        if(rgb.velocity.y < 0f && direction==jumpDirection.Up)
```